06-html5-apis/01_Canvas/js/index.js

```diff
@@ -0,0 +1,52 @@
+window.onload = function() {
+	init();
+}
+
+var rectangleX = 0;
+var smileX = 820;
+
+function init(){
+	window.requestAnimationFrame(draw);
```

Comment:
Careful here, requestAnimationFrame is not fully supported by all modern browser, you might need a workaround in a real project (https://css-tricks.com/using-requestanimationframe/).